### PR TITLE
OPENEUROPA-736: Allow sub-component loading.

### PR DESCRIPTION
**Pull request by ademarco**
This PR adds support for ECL 2.0 twig template loading by accommodating the loading process for the conventions established in https://github.com/ec-europa/ecl-twig/blob/master/docs/conventions.md

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -9,12 +9,17 @@
         "twig/twig": "^1.34"
     },
     "require-dev": {
-        "openeuropa/code-review": "~1.0",
+        "cweagans/composer-patches": "~1.6",
+        "openeuropa/code-review": "~1.0-beta2",
```
Comment (voidtek) [inline composer.json]: Is it the same of `"openeuropa/code-review": "~1.0.0-beta2"`

Comment (ademarco) [reply composer.json]: Yep, that's an old habit, sorry. I'll change it.

File: tests/EuropaComponentLibraryLoaderTest.php [hunk 2]
```diff
@@ -42,9 +40,18 @@ public function testLoader($name, $expected)
...
-            ['@namespace2/ecl-component3', 'ecl-component3.twig'],
+            // Test without prefix.
+            ['@namespace1/component1', 'component1/component1.twig', ''],
+            // Test without prefix and with inclusions.
+            ['@namespace2/component2', 'component2/component2.twig component1/component1.twig', ''],
+            ['@namespace2/component3', 'component3/component3.twig component2/component2.twig component1/component1.twig', ''],
+            // Test without prefix and sub component.
+            ['@namespace2/component4/sub-component1', 'component4/sub-component1.twig', ''],
+            // Test with prefix.
+            ['@namespace1/component5', 'ecl-component5/component5.twig', 'ecl-'],
```
Comment (voidtek) [inline tests/EuropaComponentLibraryLoaderTest.php]: For ECL 2.x, this case is not correct.
The test should be `['@namespace1/ecl-component5', 'ecl-component5/component5.twig', 'ecl-'],`.

e.g: https://github.com/ec-europa/ecl-twig/tree/master/src/ec/packages/ec-component-accordion
The name is `@ecl-twig/ec-component-accordion` (the package is `ecl-twig` and the prefix is `ec-component`)
the template path is `ec-component-accordion/accordion.html.twig`.
So the prefix is already present on the package name.

Comment (ademarco) [reply tests/EuropaComponentLibraryLoaderTest.php]: We discussed on Slack about this and clarified that the ECL repo names do not mean anything for us, hence @voidtek 's approval.

Comment (voidtek) [reply tests/EuropaComponentLibraryLoaderTest.php]: After a conversation with @ademarco, the test is correct for the future use of ecl2.

**Pull request merged**